File: research/experiments/sweep/runs/mathexpr-full-csharp-sonnet45/MathExprParser/TokenTypes.cs
namespace MathExprParser;

public enum TokenKind
{
    Number,
    Plus,
    Minus,
    Star,
    Slash,
    Percent,
    Power,
    LParen,
    RParen
}

public record Token(TokenKind Kind, string Value);

public static class TokenFactory
{
    public static Token CreateToken(TokenKind kind, string value)
    {
        return new Token(kind, value);
    }
}

File: research/experiments/sweep/runs/mathexpr-full-kotlin-sonnet45/src/main/kotlin/mathexpr/TokenTypes.kt
package mathexpr

enum class TokenKind {
    NUMBER,
    PLUS,
    MINUS,
    STAR,
    SLASH,
    PERCENT,
    POWER,
    LPAREN,
    RPAREN
}

data class Token(val kind: TokenKind, val value: String)

fun token(kind: TokenKind, value: String): Token {
    return Token(kind, value)
}
